north/src/linux/mount.rs

```diff
@@ -146,18 +146,9 @@ pub async fn mount(
 }
 
 pub async fn unmount(target: &Path) -> Result<()> {
-    let c_target = CString::new(target.as_os_str().as_bytes())?;
-
-    task::spawn_blocking(move || {
-        let result = unsafe { libc::umount(c_target.as_ptr()) };
-
-        match result {
-            0 => Ok(()),
-            _err => Err(anyhow!(
-                "Failed to unmount: {}",
-                std::io::Error::last_os_error()
-            )),
-        }
-    })
-    .await
+    let path = target.to_owned();
+    task::spawn_blocking(move || nix::mount::umount(path.as_os_str()))
+        .await
+        .map_err(anyhow::Error::new)
```

Comment:
Is this `map_err` needed here? Think the `context` call makes the error a `anyhow::Error` anyway